### PR TITLE
feat: new og images

### PR DESCRIPTION
**Pull request by harlan-zw**
### 🔗 Linked issue

https://github.com/npmx-dev/npmx.dev/pull/1654

### 🧭 Context

The previous PR was rolled back due to some Takumi stability issues. These are now solved afaik and OG Image got a [v6 stable release](https://github.com/nuxt-modules/og-image/releases/tag/v6.0.0) so it's a good time to reintroduce it. (Takumi is v1 beta now so also likely stable very soon)

:warning:  Docus < v5.9.0 uses the old version of og image which causes a version compat issue. Upgrading Docus to v5.9.0 creates a bunch of new issues that seem unrelated to og images, we should investigate that seperately.

### 📚 Description

Same as before, I've resynced the UI to match recent branding changes.

<details>
<summary>sample - check PR files for more</summary>

https://npmx.dev/
og-image-home

https://npmx.dev/accessibility
og-image-accessibility

https://npmx.dev/blog/alpha-release
og-image-blog-alpha-release

https://npmx.dev/package/@nuxt/kit
og-image-package--nuxt-kit

https://npmx.dev/package-docs/ufo/v/1.6.3
og-image-package-docs-ufo-v-1-6-3

</details>

Supersedes #2122 (rebased on latest main with CI fixes).

### CODE REVIEW

**Review by danielroe — APPROVED** [review]

File: patches/@nuxt__test-utils.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/dist/config.mjs b/dist/config.mjs
+index 41a73e237a7ecd0c63985421737f3cf0782391ee..56c9e2ca1dfade16afb10d90cebe43803063f837 100644
+--- a/dist/config.mjs
++++ b/dist/config.mjs
+@@ -128,7 +128,7 @@ async function getVitestConfigFromNuxt(options, loadNuxtOptions = {}) {
+       },
+       test: {
+         environmentOptions: {
+-          nuxtRuntimeConfig: applyEnv(structuredClone(options.nuxt.options.runtimeConfig), {
++          nuxtRuntimeConfig: applyEnv(JSON.parse(JSON.stringify(options.nuxt.options.runtimeConfig)), {
```
Comment (harlan-zw) [inline patches/@nuxt__test-utils.patch:10]: @danielroe I tried debugging what could have been causing this issue in nuxt og image (or nuxt site config) but as far as I could tell, neither of them had non-serializable data in the runtime config so it was very strange.

Do you think this fix upstream in @nuxt/test-utils makes sense? Happy to send a PR for it if so. We may consider a structuredClone with this json parse/stringify as fallback if it fails with a warning instead.

Context: https://github.com/npmx-dev/npmx.dev/actions/runs/23634594311/job/68841078028?pr=2292

File: nuxt.config.ts [hunk 2]
```diff
@@ -290,18 +289,9 @@ export default defineNuxtConfig({
...
-    },
-    fonts: [
-      { name: 'Geist', weight: 400, path: '/fonts/Geist-Regular.ttf' },
-      { name: 'Geist', weight: 500, path: '/fonts/Geist-Medium.ttf' },
-      { name: 'Geist', weight: 600, path: '/fonts/Geist-SemiBold.ttf' },
-      { name: 'Geist', weight: 700, path: '/fonts/Geist-Bold.ttf' },
-      { name: 'Geist Mono', weight: 400, path: '/fonts/GeistMono-Regular.ttf' },
-      { name: 'Geist Mono', weight: 500, path: '/fonts/GeistMono-Medium.ttf' },
-      { name: 'Geist Mono', weight: 700, path: '/fonts/GeistMono-Bold.ttf' },
-    ],
```
Comment (ghostdevv) [inline nuxt.config.ts:325]: not needed anymore?

File: nuxt.config.ts [hunk 3]
```diff
@@ -290,18 +289,9 @@ export default defineNuxtConfig({
...
+    security: {
+      maxQueryParamSize: 2048,
+    },
```
Comment (ghostdevv) [inline nuxt.config.ts:314]: oh?

File: nuxt.config.ts [hunk 4]
```diff
@@ -75,7 +75,6 @@ export default defineNuxtConfig({
           href: '/opensearch.xml',
         },
       ],
-      meta: [{ name: 'twitter:card', content: 'summary_large_image' }],
```
Comment (ghostdevv) [inline nuxt.config.ts:79]: not needed anymore?

File: test/e2e/og-image.spec.ts [hunk 5]
```diff
@@ -1,14 +1,51 @@
...
-    test(`og image for ${path}`, async ({ page, goto, baseURL }) => {
+/**
+ * OG image snapshot tests (Takumi templates).
+ *
+ * Each entry tests a different visual edge case to catch layout/overflow regressions:
+ * - Home page (Splash.takumi)
+ * - Static pages (Page.takumi)
+ * - Packages (Package.takumi with download-chart, code-tree, function-tree variants)
+ */
+const testCases = [
```
Comment (ghostdevv) [inline test/e2e/og-image.spec.ts]: could we use screaming snake case

File: uno.config.ts [hunk 6]
```diff
@@ -22,6 +22,18 @@ const customIcons = {
...
+    pipeline: {
+      exclude: [
+        // Preserve the UnoCSS defaults that @unocss/nuxt normally sets
+        /\.(css|postcss|sass|scss|less|stylus|styl)($|\?)/,
+        /\?macro=true/,
+        // Exclude OG image templates from the pipeline
+        '**/OgImage/*.takumi.vue',
+      ],
+    },
+  },
```
Comment (ghostdevv) [inline uno.config.ts:36]: could you explain how this works?

File: app/components/OgImage/Package.takumi.vue [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+          class="w-5 h-5 shrink-0 text-fg-muted force-mr-1.5 i-lucide:list"
+        />
+        <span
+          v-else-if="row.icon === 'i-lucide:package'"
+          class="w-5 h-5 shrink-0 text-fg-muted force-mr-1.5 i-lucide:package"
+        />
+        <span
+          v-else-if="row.kind === 'symbol'"
+          class="w-5 h-5 shrink-0 text-fg-muted force-mr-1.5 i-lucide:code"
+        />
```
Comment (ghostdevv) [inline app/components/OgImage/Package.takumi.vue]: does this branch need an else?

File: app/components/OgImage/Package.takumi.vue [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+          :style="{ textOverflow: 'ellipsis', lineClamp: 1, wordBreak: 'break-all' }"
+        >
+          {{ pkgNameParts.short }}
+        </div>
+        <div
+          v-if="version"
+          class="pt-3 lg:text-4xl text-3xl opacity-70 font-mono tracking-tight leading-none"
+          :style="{ textOverflow: 'ellipsis', lineClamp: 1 }"
+        >
+          v{{ version }}
```
Comment (ghostdevv) [inline app/components/OgImage/Package.takumi.vue:288]: does this display the resolved version, or could it display the range? not sure which we want here 🤔

File: app/pages/about.vue [hunk 9]
```diff
@@ -12,11 +12,14 @@ useSeoMeta({
...
-  primaryColor: '#60a5fa',
-  title: 'about npmx',
-  description: 'a fast, modern browser for the **npm registry**',
-})
+defineOgImage(
+  'Page.takumi',
+  {
+    title: () => `${$t('about.title')}`,
+    description: 'a fast, modern browser for the npm registry',
+  },
```
Comment (ghostdevv) [inline app/pages/about.vue:20]: why is this partially translated?

**Review by ghostdevv — DISMISSED** [review]

File: nuxt.config.ts (hunk 2 above)
Comment (harlan-zw) [reply nuxt.config.ts:325]: No, the latest OG Image directly integrates with Nuxt Fonts.

File: uno.config.ts (hunk 6 above)
Comment (harlan-zw) [reply uno.config.ts:36]: It's to avoid og image specific classes ending up in the main unocss css chunk. This is common due to needing hardcoded pixels in many cases to get the layout right i.e `w-[340px]`

Unfortunately, unocss doesn't seem ot merge the exclude rules with the defaults, so we have to re-implement them here. It would look a lot more straightforward otherwise (the comment covers this).

File: nuxt.config.ts (hunk 4 above)
Comment (harlan-zw) [reply nuxt.config.ts:79]: All tags needed for og images are covered by `defineOgImage()`

File: app/components/OgImage/Package.takumi.vue (hunk 8 above)
Comment (harlan-zw) [reply app/components/OgImage/Package.takumi.vue:288]: The resolved version, how would the range work? A range would seem like it would create extra scope for rendering bugs.

File: app/components/OgImage/Package.takumi.vue (hunk 7 above)
Comment (harlan-zw) [reply app/components/OgImage/Package.takumi.vue]: No, the previous spans are icons highlighting the type of `row.name`

File: app/pages/about.vue (hunk 9 above)
Comment (harlan-zw) [reply app/pages/about.vue:20]: To follow the convention that was already in place on app/pages/accessibility.vue. Originally, I removed all of these as they were redundant, as we don't have unique URLs for locales, so you'd always get the same locale's og image data. However, removing them creates errors with the i18n usage detection logic and makes the PR larger than it already is, so we keep them for now to avoid these.

File: test/e2e/og-image.spec.ts (hunk 5 above)
Comment (harlan-zw) [reply test/e2e/og-image.spec.ts]: Updated, though this is a nitpick as it follows previous conventions from the file.

File: nuxt.config.ts (hunk 3 above)
Comment (harlan-zw) [reply nuxt.config.ts:314]: Care to elaborate? For some extra context, the latest og image has a `strict` mode, which I've updated the code for and we leverage the existing `NUXT_IMAGE_PROXY_SECRET` for signed og image URLs (avoids tampering).

**Comment by harlan-zw** [comment]
Just a heads up on this, I can't spend any more of my time resolving the ongoing conflicts. If you're interested in this work, then let's merge and improve anything else in subsequent PRs; otherwise, let's close it off.

File: nuxt.config.ts [hunk 10]
```diff
@@ -128,14 +127,7 @@ export default defineNuxtConfig({
...
-    '/__og-image__/**': getISRConfig(3600),
-    '/__og-image__/image/compare/**': {
-      isr: {
-        expiration: 3600,
-        passQuery: true,
-        allowQuery: ['packages', '_query'],
-      },
-    },
+    '/_og/d/**': getISRConfig(60 * 60 * 24), // 1 day
     '/_avatar/**': { isr: 3600, proxy: 'https://www.gravatar.com/avatar/**' },
```
Comment (alexdln) [inline nuxt.config.ts]: Looks like it still needs allowQuery for this route

`/_og/d/a_Compare+npm+packages+side-by-side,c_Compare.takumi,q_eyJwYWNrYWdlcyI6InJlYWN0LHR5cGUifQ,props_eyJwYWNrYWdlcyI6WyJyZWFjdCIsInR5cGUiXX0,emptyDescription_Compare+npm+packages+side-by-side,title_Compare+react+vs+type+-+npmx,description_~U2lkZS1ieS1zaWRlIGNvbXBhcmlzb24gb2YgcmVhY3QsIHR5cGU,p_Ii9jb21wYXJlIg.png?_query={"packages":"react,type"}`

File: app/composables/useCharts.ts [hunk 11]
```diff
@@ -66,6 +66,138 @@ function mergeDailyPoints(points: DailyRawPoint[]): DailyRawPoint[] {
...
+    .sort((a, b) => a.day.localeCompare(b.day))
+    .map(item => {
+      const dayDate = parseIsoDate(item.day)
+      const timestamp = dayDate.getTime()
+
+      return { day: item.day, value: item.value, timestamp }
+    })
+}
+
+export function buildRollingWeeklyEvolutionFromDaily(
```
Comment (alexdln) [inline app/composables/useCharts.ts]: Looks like this logic is not used anywhere. Can you check please and remove if so (same for monthly, yearly, daily)

File: app/components/OgImage/Package.takumi.vue
Comment (alexdln) [inline app/components/OgImage/Package.takumi.vue:1]: Can you please add the number of downloads, because right now the graph shows a trend, but not a specific value (_which is important_)

File: nuxt.config.ts (hunk 10 above)
Comment (harlan-zw) [reply nuxt.config.ts]: Hm this was actually conflicting with og images native caching so we can remove it

Comment (alexdln) [reply nuxt.config.ts]: Sorry, the thread didn't show the changes, I meant this part about passing allowQuery to compare og-image

Image

Comment (harlan-zw) [reply nuxt.config.ts]: Yup, so we can re-implement but og image internally already provides ISR headers based on the resolved request, so it should be behaving the same. Let me confirm the logic for this, though.

File: app/components/OgImage/Package.takumi.vue
Comment (harlan-zw) [reply app/components/OgImage/Package.takumi.vue:1]: I have re-added that, having some issues regenerating the snapshots currently with the downloads. Will return to this in a few hours.

File: app/composables/useCharts.ts (hunk 11 above)
Comment (harlan-zw) [reply app/composables/useCharts.ts]: Ah, yes, good catch, looks like it got superseded in a recent change. Have removed.

File: app/components/OgImage/Package.takumi.vue [hunk 12]
```diff
@@ -184,8 +189,23 @@ const fetchLikes = $fetch<{ totalLikes: number }>(`/api/social/likes/${name}`)
...
 
+const downloadUrl = `https://api.npmjs.org/downloads/point/last-week/${encodePackageName(name)}`
+const fetchDownloads = $fetch<{ downloads: number }>(downloadUrl)
+  .then(d => {
+    console.error('[og-image-package] downloads response:', JSON.stringify(d))
+    weeklyDownloads.value = d?.downloads ?? 0
+  })
+  .catch(err => {
+    console.error('[og-image-package] downloads fetch failed:', downloadUrl, err?.message || err)
+  })
```
Comment (alexdln) [inline app/components/OgImage/Package.takumi.vue]: Is this different with the last point in weeklyValues?

Comment (harlan-zw) [reply app/components/OgImage/Package.takumi.vue]: Cleaned that up :+1:

File: app/components/OgImage/Package.takumi.vue
Comment (harlan-zw) [reply app/components/OgImage/Package.takumi.vue:1]: Snapshots updated

File: nuxt.config.ts (hunk 10 above)
Comment (alexdln) [reply nuxt.config.ts]: Were you able to find the information? This logic, as far as I know, is specific to Vercel and is difficult to replicate elsewhere (_including in previews_). And without allowQuery, they [query params] simply might not be passed to components

Comment (harlan-zw) [reply nuxt.config.ts]: Internally, OG Image has [caching on the images themselves](https://github.com/nuxt-modules/og-image/blob/main/src/runtime/server/util/eventHandlers.ts#L136), ensuring that regardless of what the CDN layer is doing, the images get cached to avoid expensive re-renders. This uses the default Nitro cache, which, as far as I can tell, is set up in npmx.dev for `vercel-runtime-cache`.

Additionally, og images will emit two layers of headers, the original route rules on the dynamic and static endpoints at https://github.com/nuxt-modules/og-image/blob/main/src/module.ts#L884 and specific headers on the response itself it the route rules aren't acknowledged for whatever reason https://github.com/nuxt-modules/og-image/blob/945bdb9b20209abbd29d737ef3594ba97c21d754/src/runtime/server/util/cache.ts#L107.

The query params are irrelevant due to the query not affecting the image, as the props are passed in as the regular URL params, i.e `/_og/d/q_dynamic-query`. The query param does end up in the query, but shouldn't be included as a cache vary, as if the query doesn't affect the props, it's irrelevant. This was an improvement for v5 -> v6 as the caching is more intuitive for CDNs.

With the og image secret passed, we also encode all URLs, which would affect the query param collapsing.

Saying all that, testing across environments and at these different levels gets confusing, so we should of course, double-check it's working as intended after deployment.

**Comment by harlan-zw** [comment]
> ## ⚠️ Dependency Count
> This PR adds 30 new dependencies (2271 → 2301), which exceeds the threshold of 15.

This is out of date, the issue was docus was using v5 nuxt-og-image so we were paying double version dependency cost. Also we get hit with the storybook deps for some reason, guessing because we had to lint the lock.

We disable og image in docus for the time being as there's another issue with latest that needs to be investigated seperately.

```
Run e18e/action-dependency-diff@f825d5b5c5ce0a42dc46c47ec20de24460affcd8
Workspace path is /home/runner/work/npmx.dev/npmx.dev
Detected lockfile pnpm-lock.yaml
Using lockfile: pnpm-lock.yaml
Comparing package lockfiles between origin/main and e59f10ea5a6eb86f40cbd9691ebaaa0fb469ee88
Parsed current lockfile with 2308 packages
Parsed base lockfile with 2274 packages
Dependency threshold set to 15
Size threshold set to 200 kB
Duplicate threshold set to 4
Pack size threshold set to 50 kB
Base dependency count: 2271
Current dependency count: 2301
Dependency count increase: 30
Found 31 new package versions
Found 27 removed package versions.
Added 70333 bytes for @takumi-rs/core@1.0.5
Added 5216932 bytes for @takumi-rs/wasm@1.0.5
Added 20683 bytes for defu@6.1.7
Added 2843696 bytes for nuxt-og-image@6.3.10
Added 185343 bytes for @storybook/addon-a11y@10.3.5
Added 1663357 bytes for @storybook/addon-docs@10.3.5
Added 17963 bytes for @storybook/addon-themes@10.3.5
Added 48793 bytes for devalue@5.7.1
Added 35426 bytes for tinyexec@1.1.1
Added 38673 bytes for tinyglobby@0.2.16
Added 328752 bytes for @eslint/config-array@0.23.5
Added 67104 bytes for @eslint/object-schema@3.0.5
Added 58043 bytes for @eslint/config-helpers@0.5.5
Added 92054 bytes for @eslint/core@1.2.1
Added 108176 bytes for @eslint/plugin-kit@0.7.1
Added 6249953 bytes for @napi-rs/wasm-runtime@1.1.3
Added 109262 bytes for @nuxt/devtools-kit@4.0.0-alpha.3
Added 1380331 bytes for oxc-parser@0.124.0
Added 1408380 bytes for nuxt-site-config@4.0.8
Added 7398 bytes for @storybook/csf-plugin@10.3.5
Added 18620 bytes for @storybook/react-dom-shim@10.3.5
Added 54571 bytes for @takumi-rs/helpers@1.0.5
Added 6928223 bytes for @takumi-rs/core-linux-x64-gnu@1.0.5
Added 44501 bytes for @oxc-project/types@0.124.0
Added 24089 bytes for call-bind@1.0.9
Added 2787848 bytes for es-abstract@1.24.2
Added 1108305 bytes for culori@4.0.2
Added 48046 bytes for nuxtseo-shared@5.1.3
Added 25324 bytes for nuxt-site-config-kit@4.0.8
Added 27426 bytes for site-config-stack@4.0.8
Added 2593415 bytes for @oxc-parser/binding-linux-x64-gnu@0.124.0
Subtracted 3076931 bytes for nuxt-og-image@5.1.13
Subtracted 185343 bytes for @storybook/addon-a11y@10.3.4
Subtracted 1663357 bytes for @storybook/addon-docs@10.3.4
Subtracted 17969 bytes for @storybook/addon-themes@10.3.4
Subtracted 320890 bytes for @eslint/config-array@0.23.4
Subtracted 59201 bytes for @eslint/object-schema@3.0.4
Subtracted 55761 bytes for @eslint/config-helpers@0.5.4
Subtracted 54158 bytes for @eslint/core@1.2.0
Subtracted 93553 bytes for @eslint/plugin-kit@0.7.0
Subtracted 325374 bytes for execa@9.6.1
Subtracted 5224260 bytes for satori@0.18.4
Subtracted 7398 bytes for @storybook/csf-plugin@10.3.4
Subtracted 18620 bytes for @storybook/react-dom-shim@10.3.4
Subtracted 7619 bytes for is-stream@4.0.1
Subtracted 2786378 bytes for es-abstract@1.24.1
Subtracted 28353 bytes for get-stream@9.0.1
Subtracted 28040 bytes for human-signals@8.0.1
Subtracted 4382 bytes for strip-final-newline@4.0.0
Subtracted 36841 bytes for figures@6.1.0
Subtracted 14213 bytes for pretty-ms@9.3.0
Subtracted 10509 bytes for yoctocolors@2.1.2
Subtracted 3958 bytes for is-unicode-supported@2.1.0
Subtracted 15563 bytes for @sec-ant/readable-stream@0.4.1
Subtracted 378406 bytes for image-size@2.0.2
Subtracted 7679 bytes for satori-html@0.3.2
Subtracted 289748 bytes for yoga-wasm-web@0.3.3
Subtracted 4311 bytes for parse-ms@4.0.0
Total dependency size increase: 18.9 MB
Wrote artifact to /home/runner/work/npmx.dev/npmx.dev/.e18e-tmp/e18e-diff-result.json
```

**Review by alexdln — APPROVED** [review]
lgtm, thanks for updates and replies 🤍

File: nuxt.config.ts (hunk 3 above)
Comment (ghostdevv) [reply nuxt.config.ts:314]: I have no idea what I was referring to here, so we can leave it - but ty!

File: app/components/OgImage/BlogPost.takumi.vue [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+  date = '',
+} = defineProps<{
+  title: string
+  authors?: ResolvedAuthor[]
+  date?: string
+}>()
+
+const formattedDate = computed(() => {
+  if (!date) return ''
+  try {
```
Comment (ghostdevv) [inline app/components/OgImage/BlogPost.takumi.vue]: does this need a try catch?

File: app/pages/package/[[org]]/[name].vue [hunk 14]
```diff
@@ -2,12 +2,6 @@
 import { assertValidPackageName } from '#shared/utils/npm'
 import { getDependencyCount } from '~/utils/npm/dependency-count'
 
-defineOgImageComponent('Package', {
-  name: () => packageName.value,
-  version: () => requestedVersion.value ?? '',
```
Comment (ghostdevv) [inline app/pages/package/[[org]]/[name].vue:7]: was removing the `?? ''` intentional?

File: app/composables/useCharts.ts [hunk 15]
```diff
@@ -66,6 +66,43 @@ function mergeDailyPoints(points: DailyRawPoint[]): DailyRawPoint[] {
     .map(([day, value]) => ({ day, value }))
 }
 
+/** Catmull-Rom monotone cubic spline — same algorithm as vue-data-ui's smoothPath for OG Images */
+export function smoothPath(pts: { x: number; y: number }[]): string {
```
Comment (ghostdevv) [inline app/composables/useCharts.ts:74]: there are some oxlint issues here

File: app/components/OgImage/Package.takumi.vue [hunk 16]
```diff
@@ -0,0 +1,416 @@
...
+
+// Flatten file tree into renderable rows for code-tree variant
+interface TreeRow {
+  name: string
+  depth: number
+  isDir: boolean
+}
+const treeRows = shallowRef<TreeRow[]>([])
+
+async function fetchCodeTree() {
```
Comment (ghostdevv) [inline app/components/OgImage/Package.takumi.vue:124]: some oxlint issues here

File: test/fixtures/mock-routes.cjs [hunk 17]
```diff
@@ -181,10 +181,28 @@ function matchNpmApi(urlString) {
...
+    const packageName = rangeMatch[2]
+    const [start, end] = dateRange.split(':')
+
+    // Generate deterministic daily download data from the package name
+    const seed = packageName.split('').reduce((s, c) => s + c.charCodeAt(0), 0)
+    const base = (seed % 9000) + 1000
+    const downloads = []
+    const startDate = new Date(start)
+    const endDate = new Date(end)
+    for (let d = new Date(startDate); d <= endDate; d.setUTCDate(d.getUTCDate() + 1)) {
```
Comment (ghostdevv) [inline test/fixtures/mock-routes.cjs]: some oxlint issues here

File: modules/runtime/server/cache.ts [hunk 18]
```diff
@@ -187,6 +187,46 @@ function getMockForUrl(url: string): MockResult | null {
...
+      // Wave period varies per package (20-60 days)
+      const wavePeriod = 20 + ((s >> 8) % 40)
+      const waveAmp = 0.1 + ((s >> 12) % 30) / 100 // 0.10 .. 0.40
+      // Weekend dip intensity
+      const weekendDip = 0.3 + ((s >> 16) % 40) / 100 // 0.30 .. 0.70
+
+      const downloads: { day: string; downloads: number }[] = []
+      const start = new Date(startDate!)
+      const end = new Date(endDate!)
+      for (let d = new Date(start); d <= end; d.setUTCDate(d.getUTCDate() + 1)) {
```
Comment (ghostdevv) [inline modules/runtime/server/cache.ts]: some oxlint issues here

File: test/nuxt/components/OgImagePackage.spec.ts [hunk 19]
```diff
@@ -136,25 +135,14 @@ describe('OgImagePackage', () => {
...
-      setupMocks({ totalLikes: 42 })
-
-      const component = await mountSuspended(OgImagePackage, {
-        props: baseProps,
-      })
-
-      expect(component.text()).toContain('42')
-    })
+    expect(component.text()).toContain('test')
+    expect(component.text()).toContain('repo')
```
[the remark this answers is not in the document]
Comment (ghostdevv) [reply test/nuxt/components/OgImagePackage.spec.ts]: this seems valid

File: app/components/OgImage/Profile.takumi.vue [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+  avatar = '',
+  packageCount = 0,
+  orgCount = 0,
+} = defineProps<{
+  handle: string
+  displayName?: string
+  description?: string
+  avatar?: string
+  packageCount?: number
+  orgCount?: number
```
Comment (ghostdevv) [inline app/components/OgImage/Profile.takumi.vue]: are these used? or future proofing?

File: uno.config.ts [hunk 21]
```diff
@@ -22,6 +22,18 @@ const customIcons = {
 }
 
 export default defineConfig({
+  // og-image uses hardcoded classes we don't want bundled into main app
+  content: {
+    pipeline: {
+      exclude: [
+        // Preserve the UnoCSS defaults that @unocss/nuxt normally sets
```
Comment (ghostdevv) [inline uno.config.ts:29]: how come they're not being set

**Review by ghostdevv — COMMENTED** [review]
a few more things 😅 code rabbit also has a couple things unresolved

File: app/composables/useCharts.ts (hunk 15 above)
Comment (harlan-zw) [reply app/composables/useCharts.ts:74]: Ah I didn't see the warnings*

File: app/components/OgImage/BlogPost.takumi.vue (hunk 13 above)
Comment (harlan-zw) [reply app/components/OgImage/BlogPost.takumi.vue]: Usually a good idea with date parsing

File: app/pages/package/[[org]]/[name].vue (hunk 14 above)
Comment (harlan-zw) [reply app/pages/package/[[org]]/[name].vue:7]: Yes, null should be passed for a missing version not an empty string (which is handled by Nuxt islands automatically when nullish value is passed).

File: uno.config.ts (hunk 21 above)
Comment (harlan-zw) [reply uno.config.ts:29]: Because we're redefining the config, you'd have to look at the unocss module internals to know more.

File: app/components/OgImage/BlogPost.takumi.vue (hunk 13 above)
Comment (harlan-zw) [reply app/components/OgImage/BlogPost.takumi.vue]: Removed it

File: patches/@nuxt__test-utils.patch (hunk 1 above)
Comment (harlan-zw) [reply patches/@nuxt__test-utils.patch:10]: Going to just resolve this for now - easy to fix later and doesn't cause issues.

File: app/components/OgImage/Profile.takumi.vue (hunk 20 above)
Comment (harlan-zw) [reply app/components/OgImage/Profile.takumi.vue]: Good catch, have updated those. I'm also open to dropping this og image
